tests/puppeteer/campaigns/functional/BO/orders/invoices/01_generateInvoiceByDate.js

```diff
@@ -0,0 +1,133 @@
+        await expect(pageTitle).to.contains(this.pageObjects.viewOrderPage.pageTitle);
+      });
+
+      it('should change the order status to \'Shipped\' and check the validation', async function () {
+        const result = await this.pageObjects.viewOrderPage.modifyOrderStatus(Statuses.shipped.status);
+        await expect(result).to.be.true;
+      });
+
+      it('should go to the orders page', async function () {
+        await this.pageObjects.boBasePage.goToSubMenu(
+          this.pageObjects.boBasePage.ordersParentLink,
+          this.pageObjects.boBasePage.ordersLink,
+        );
+        const pageTitle = await this.pageObjects.ordersPage.getPageTitle();
+        await expect(pageTitle).to.contains(this.pageObjects.ordersPage.pageTitle);
+      });
+
+      it('should go to the second order', async function () {
+        await this.pageObjects.ordersPage.goToOrder(2);
+        const pageTitle = await this.pageObjects.viewOrderPage.getPageTitle();
+        await expect(pageTitle).to.contains(this.pageObjects.viewOrderPage.pageTitle);
+      });
+
+      it('should change the order status to \'Payment accepted\' and check the validation', async function () {
+        const result = await this.pageObjects.viewOrderPage.modifyOrderStatus(Statuses.paymentAccepted.status);
+        await expect(result).to.be.true;
+      });
+    });
+
+    describe('Generate PDF file by date', async () => {
+      it('should go to invoices page', async function () {
+        await this.pageObjects.boBasePage.goToSubMenu(
+          this.pageObjects.boBasePage.ordersParentLink,
+          this.pageObjects.boBasePage.invoicesLink,
+        );
+        const pageTitle = await this.pageObjects.invoicesPage.getPageTitle();
+        await expect(pageTitle).to.contains(this.pageObjects.invoicesPage.pageTitle);
+      });
+
+      it('should generate PDF file by date and check the file existence', async function () {
+        await this.pageObjects.invoicesPage.generatePDFByDate();
+        const exist = await this.pageObjects.boBasePage.checkFileExistence(
+          global.BO.DOWNLOADSPATH,
+          Invoices.moreThanAnInvoice.fileName,
+        );
+        await expect(exist).to.be.true;
+      });
+
+      it('should delete the invoice', async () => {
+        await files.deleteFile(`${global.BO.DOWNLOADSPATH}/${Invoices.moreThanAnInvoice.fileName}`);
+        const exist = await files.isFileExist(
+          global.BO.DOWNLOADSPATH,
+          Invoices.moreThanAnInvoice.fileName,
+        );
+        await expect(exist).to.be.false;
+      });
+
+      it('should check the error message when there is no invoice in the entered date', async function () {
+        await this.pageObjects.invoicesPage.generatePDFByDate(dateFrom, dateTo);
+        const textMessage = await this.pageObjects.invoicesPage.getTextContent(
+          this.pageObjects.invoicesPage.alertTextBlock,
+        );
```

Comment:
You should return message in function `generatePDFByDate`

Reply:
The message is not displayed all the time.